logstash-core/lib/logstash/instrument/collector.rb

```diff
@@ -48,9 +48,16 @@ def initialize
     def push(namespaces_path, key, type, *metric_type_params)
       begin
         metric_proxy = get(namespaces_path, key, type)
-        return metric_proxy.execute(*metric_type_params) if metric_proxy.respond_to?(:execute)
-
-        logger.error("Collector: Cannot record metric action #{type}@#{metric_type_params.join('/')} on <#{metric_proxy}> at path #{namespaces_path.join('/')}/#{key}")
+        expended_params = *metric_type_params
+        case type
+        when :gauge
+          return metric_proxy.set(expended_params[1])
+        when :counter
+          return metric_proxy.increment if expended_params[1] == nil
+          return metric_proxy.increment(expended_params[1])
+        else
```

Comment:
minor code style suggestion, but not necessary:

```suggestion
        metric_type, metric_arg = *metric_type_params
        case metric_type
        when :gauge
          return metric_proxy.set(metric_arg)
        when :counter
          return metric_proxy.increment if metric_arg.nil?
          return metric_proxy.increment(metric_arg)
        else
```

Reply:
Why `expended_params[0]` has also the same value of `type`?

For example https://github.com/elastic/logstash/blob/6b8d0903ce372f061f05571cdb210b0a95416948/logstash-core/src/main/java/org/logstash/instrument/metrics/MetricExt.java#L97

invokes `push` with (namespace, key, "gauge", "increment", value), to me `expended_params[0]` is "increment" and not "gauge"